### PR TITLE
Removes bolt of change from magicarps

### PR DESCRIPTION
**Pull request by improvedname**
## About The Pull Request

Removes bolt of change from magicarps and chaos carps.

## Why It's Good For The Game
Because i consider the  ## bolt of change carp the cheese carp
Because lets say a threat poses on the station like a ninja. or something with non strippable items due its locked to them. just get a carp and he gets stripped wholely. no counter play can't even block the shot with the fucking katana? miner unsleashes megafauna onboard the station? magicarp no counterplay. Magic can't be blocked besides anti-magic items((not like anyone bothers getting them in a non wizard/cult excepts chaplains/miners round due it being a normal round without magic))

Bolt of change currently allows for the following cheese which i think is bad.
You can rush rainbow slimes by changing some random mob rapidly((can even be a slaved unit))
You can make a army of lesser drakes because fuck you((you can argue that its RNG based but bro rng isn't that hard when its only a tiny list bruh...))
You can transform anything in a syndicate borg and if you deconstruct/transform them they drop a syndikey headset which cheeses and potentionally fuck over antags by just doing your normal job(syndikey should be received either trough a event/spaceruin/or auctally catching a fucking agent, just because somoene was doing xenobio shouldnt have let security get all syndikeys cuz fuck your antag round and auctally communicating to other traitors right?)
 
## Changelog
:cl:
del: Removes magicarp bolt of change
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
remove the whole carp

**Comment by improvedname** [comment]
tl:dr i ded

**Comment by trollbreeder** [comment]
i ded pls merge pr

**Comment by Shadowflame909** [comment]
Well this PR is gonna be merged because a maintainer has been gunning and waiting for it since forever.

My only thing is, your reasoning is shit because a little thing called holy melons exist. They are anti magic melons you can easily give out to the entire crew.

Now, it's time for me to start working on the staff of change replacement item.

**Comment by improvedname** [comment]
> Well this PR is gonna be merged because a maintainer has been gunning and waiting for it since forever.
> 
> My only thing is, your reasoning is shit because a little thing called holy melons exist. They are anti magic melons you can easily give out to the entire crew.
> 
> Now, it's time for me to start working on the staff of change replacement item.

Yeah but do you ever see xenobiologists making those when it auctally happens?

**Comment by Shadowflame909** [comment]
Bro

Those are a botanist item.

Also, staff of change is a beneficial item most of the crew is looking forward to. So a botanist heeds not to make something to counter it.

If you really wanted more ninja power-creep. You'd just give the ninja hardsuit anti-magic.

What a shame that space-ninja power-creep is what the greatest xenobiology power dies on.

**Comment by improvedname** [comment]
yeah staff of change is so benifictal it can cuck over any antag without counterplay you a self reviving changeling? too bad ur a mouse now lol

**Comment by Shadowflame909** [comment]
In terms of yeeting antag potential. If you replace a Mech with magicarp of change in this sentence. Nothing would change.

Mechs can gib lings bro.

**Comment by trollbreeder** [comment]
might as well make a new slime crossbreed that polymorphs the user

**Comment by Shadowflame909** [comment]
@trollbreeder 

The solution is just rainbow and gold.

Self-Polymorph is in a nerf in of itself, because you cannot change what you get. Since you lose hands by becoming a simple mob.

**Comment by Ratvar** [comment]
What I see is shadowflame literally stating that holy mellons should be removed because they cuck all cults and wizard gamemode with 0 counterplay
He alsy says that mechs need a nerf too

**Comment by improvedname** [comment]
Shadow bruh the normal wizard magicarps are untouched the xenocarps are a subtype @Shadowflame909

**Comment by Shadowflame909** [comment]
The god of clock cult would see things that way yes.

Nerfing things because they killed you, and not because they made the game less fun is in bad taste.

**NERFING** what's supposed to be the pinnacle of said department's destructive power for better or worse. Because the magnum opus could stand up to antagonists. Is in bad taste in it of itself.

Why do we even have antagonists with the intent to challenge the pinnacle of departments?

Like the glass cannon ninja, with the strongest melee weapon in the game, stun gloves that got removed from the traitor set for being too op, meth speed active all the time, and an emag embedded into the said weapon.

If we don't want them standing up to peak tech?

The same thing can be said for the random 1-hour event slaughter demons, the space dragon event, and xenomorphs. Who are quoted to be the pinnacle of evolution versus technology?


It's a literal arms race, and by lowering the glass ceiling of the non-antags. You're just saying, "Yes antags should always be better than the crew for no reason."

**Review by Shadowflame909 — COMMENTED** [review]
You removed it from both chaos and regular magicarps

**Comment by improvedname** [comment]
And this wasnt because ninja powercreep this is because magic of change are the bane of whafs balanced free emagg/syndi headset keys/lesser ash drakes/faster slime rushing and not that big but you can also use it to farm xenofarms

**Comment by improvedname** [comment]
Organ xeno farming*

**Comment by improvedname** [comment]
So yeah you can give everyone a hive node to be anti-facehuggable before real xenos even come

**Comment by Shadowflame909** [comment]
You can also give everyone a Mech before the xenomorphs come.

But this doesn't happen, do you know why?

The game set on paranoia wouldn't promote cooperation or interaction for this to ever happen!

Not one round has such a thing happened.

**Comment by improvedname** [comment]
Except roboticists already build a mech when theh already can?

**Comment by improvedname** [comment]
Also bro wizard magicarps are untouched again.
THESE are subtypes that only is available trough xenobiology

**Comment by Tlaltecuhtli** [comment]
why would you even farm xeno organs when you can be one or a syndiborg

**Comment by improvedname** [comment]
Im saying if yout a smart brain brain you get a golem candidate to allow this powercreep for yourself...

**Comment by Shadowflame909** [comment]
I've been arguing about this since 2014. So better not change minds with words.

The only thing that gets merged is what someone's coded improvedname. So doubt not that this PR won't go through even if I complain or not.

But do beget this. You're going to end up facing the pinnacle of Xenobio one way or another. So don't expect your space ninja rounds to ever be easy as long as that lesser ash drake is staring you down.

**Comment by AWalton** [comment]
What Shadowflame said, this is vastly overstating the actual effectiveness of magicarps vs your precious antag round. Any robust ninja has the speed to run circles around a magicarp and one (or is it two?) shot them to death. 

I ded.

**Comment by improvedname** [comment]
Not forgetting this is available in all rounds with any antags.

**Comment by IndieanaJones** [comment]
If this were to go through, you'd ought to just remove magicarps from xenobio instead.  Life and death bolts are gone, so if they lose transform, they're pointless.

**Comment by improvedname** [comment]
> If this were to go through, you'd ought to just remove magicarps from xenobio instead. Life and death bolts are gone, so if they lose transform, they're pointless.
> 
> That being said, I really, really hope this doesn't go through.

so everything needs to be about instakill? instaheal? power mob changing?
100% disknowledging the fireball(which is explosion + knockdown) the arcane bolt? the door spell? the teleport spell? the dismember spell?

**Comment by cacogen** [comment]
the unending ebb and flow of fun shit being added and nerfed or removed

**Comment by ATH1909** [comment]
Woah, woah, woah, woah, woah. There's no need to take drastic measures here, improvedname. We can reach a compromise while still fixing the issues you have gripes with.

> Because lets say a threat poses on the station like a ninja. or something with non strippable items due its locked to them. just get a carp and he gets stripped wholely. 

Pro-tip: If you see a magicarp roaming the halls as a ninja, you can just... run away. You can become nigh-invisible at-will and run at sanic speeds; you can CHOOSE YOUR BATTLES. Run off, find the chaplain, gank him, and come back with his null rod to murder that magicarp. A magicarp is arguably less threatening as a guy who's waiting for you with a shotgun out, as you can at least become immune to the strongest attack of the former.

Still, if you truly believe that change magicarps are the one true hard counter to ninjas that can't be played around in any way, I can make energy katanas make you magic immune while held (I can add a line to the description of the katana that'd say that the katana has been etched with Space Japanese prayers on its handle, has been passed down through your family for generations, and contains the spirit of your great grandpappy or something to justify it).

As for abductors, they really shouldn't be trying to fight ANY kind of simplemob, let alone magicarps. Their tools are geared towards the abduction of humanoids, not towards killing simplemobs and borgs.

> no counter play can't even block the shot with the fucking katana?

If you'd like, I can add a special exception to blocking code that'd make magic projectiles count as having 0 armor_piercing (instead of 100) for the purposes of blocking them with shields and the like (they'd still have 100 armor_piercing for the purposes of actual armor damage reduction calculations, of course).

> miner unsleashes megafauna onboard the station? magicarp no counterplay.

I can make megafauna (other than lesser ash drakes) immune to change bolts, if you'd like. That sounds like something that should be on their list of immunities.

> Magic can't be blocked besides anti-magic items((not like anyone bothers getting them in a non wizard/cult excepts chaplains/miners round due it being a normal round without magic))

But it CAN be DODGED; magicarps have a pretty slow rate of fire- they can only fire one bolt off every 3 seconds, so you can just wait for them to blow their load, move to the side to dodge the one bolt they fired, and then rip them to pieces with ranged weaponry (or a katana that can one-shot them). I can make it even slower if you think that magicarps can still fire too quickly.

> You can rush rainbow slimes by changing some random mob rapidly((can even be a slaved unit))

If you're at the point where you have both gold slime cores and a way to control the magicarps you've created (either light pink or rainbow slime cores), you can probably already make rainbow cores via the normal method. Also, if you're shooting NPC mobs to try to roll a specific kind of slime, be warned that those NPCs can turn into a very strong hostile mob and attack you/your carp (if you don't have something like that specific crossbreed that enforces a zone of peace, and I forget if your pet carp firing bolts will break that peace), and your carp only has 50 health.

> You can make a army of lesser drakes because fuck you((you can argue that its RNG based but bro rng isn't that hard when its only a tiny list bruh...))

The list is weighted; you only have a 1/200 chance (per shot) to turn someone into a lesser ash drake with a change carp bolt. A magicarp can only fire a change bolt once every 3 seconds, so it'd take 10 minutes on average of doing nothing but waiting in front of a change carp, getting bolted to turn into an ash drake.

Oh, but that isn't factoring in that weeping angels are in the pool of change bolt forms, and they're unpolymorphable, unable to move while being seen, and unable to speak. They're also a 1/200 chance per bolt, so each of your subjects has a 50% chance to end up as a weeping angel instead of an ash drake (fun fact: IIRC, weeping angels can count as seeing EACH OTHER (paralyzing both angels) if they're not in darkness, so they REALLY don't work well in teams together).


> You can transform anything in a syndicate borg and if you deconstruct/transform them they drop a syndikey headset which cheeses and potentionally fuck over antags by just doing your normal job(syndikey should be received either trough a event/spaceruin/or auctally catching a fucking agent, just because somoene was doing xenobio shouldnt have let security get all syndikeys cuz fuck your antag round and auctally communicating to other traitors right?)

I'll try to see if I can whip up a PR later tonight that'll make those keys NOT drop upon the polymorphing of a borg, because that is a known unintended feature, IIRC.

> yeah staff of change is so benifictal it can cuck over any antag without counterplay you a self reviving changeling? too bad ur a mouse now lol

I do agree that staves of change can fuck over changelings a bit too hard. Tell ya what, I'll make it so that changelings are immune to the normal effects of change bolts while they're in human form, and will instead have their human appearance randomized upon being hit by a change bolt (to make it look like they just rolled the "human" result so that they won't be outed upon being hit by a change bolt). They'll still be able to be change bolted while in monkey or headslug form, though, in case they DO want to get polymorphed (and because headslugs are in the change bolt pool).

**Comment by improvedname** [comment]
Im just saying there is no real way to tell whatever bolt the carp is. or if its hostile. the element of a sudden bolt of change fucks you up

**Comment by ATH1909** [comment]
> Im just saying there is no real way to tell whatever bolt the carp is. or if its hostile. the element of a sudden bolt of change fucks you up

Then just treat all magicarps like they have a bolt that could potentially fuck you up.

**Comment by improvedname** [comment]
except they usually have a gold reproduction extract so its infinite tries anyways

**Comment by ATH1909** [comment]
In short, this PR is a bit of a drastic measure, and I think that it'd be best to just fix some of these problems with change carps instead of taking the nuclear/no fun allowed option of removing change carp altogether.

Also, a favored pasttime of mine as a syndicate bioweapons agent is to farm the life reaction until I get a change carp, then do cool things with the forms that I can roll from it. It's actually more difficult to do than you'd think, since some of the forms in the gold slime core pool are quite tanky/strong (and they're hostile to you when you make them with the life reaction), even when you have a stetchkin and a sniper rifle. I'd hate to see cool things like that go because someone got change bolted once as a ninja because they recklessly charged a change carp.

> except they usually have a gold reproduction extract so its infinite tries anyways

How often do you actually see sapient change carps? They're quite hard to get, even for an experienced xenobiologist who can get "infinite tries" with a gold reproduction extract. You still have to tech up to both light pinks and golds in the first place.

**Comment by AdamElTablawy** [comment]
While being changebolted by a magicarp is annoying, it''s rare enough that nobody rages that their antag round was ruined by spontaneous animalification, and it's got a fair chance at backfiring. One moment you have a ninja, now you have a teleporting free drone and that's even worse.

I wouldn't remove it. There are alternatives. You could make it a delayed change that gives the person a transformation disease that they can cure instead of removing it. This leaves the option open to farm xeno organs and make cool shit without removing something fun and funny, while also not making it fuck over unwilling people quite as hard.

**Comment by FloranOtten** [comment]
TL;DR I ded.

Yeah. It fucks over antags. So does a shotgun. A large grenade is an instant gib. If you have the time to farm xenobio for a sentient magicarp, you have the time to get a shotgun.

**Comment by trexter555** [comment]
the syndieborgs dropping syndie keys are an actual issue though, if someone could fix it so that they dont drop them that'd be nice, other than that, average ided pr

**Comment by Whoneedspacee** [comment]
all these seem like very minimal easy fixes and removes a fun thing from the game for not much gain

make the projectile blockable / reflectable and make it not work on megafauna

ezpz

**Comment by XDTM** [comment]
Completely in favor of the removal. Even just in combat it's near-equivalent to a one shot kill; and we removed stuns for a similar reason (you can "just avoid them", but you can say that about taser shots and look where they are now). Bolts of change tend to be more permanent.
On top of that they have the incredible potential of acquiring antag stuff like xeno organs and syndikeys that are supposed to, y'know, only be available if you fight said antag.

Plus, i'm against stuff that heavily breaks the theme of the department and goes the lazy way of just grabbing a bunch of existing stuff at random. Which is why i removed the random species green slime and replaced it with the jellypeople subspecies. Gold slimes and magicarps by extension are another instance of that design pattern.

**Comment by Tharcoonvagh** [comment]
No fun gang
Oranges will request a policy thread anyways

**Comment by Spepbuthewranglesgithub** [comment]
bolt of death.

**Comment by trollbreeder** [comment]
bolt of death is different because it kills you instantly with no exceptions
bolt of change still lets you fight back even though you've been turned into god-knows-what and has !!FUN!! factor

**Comment by ATH1909** [comment]
> (you can "just avoid them", but you can say that about taser shots and look where they are now)

You could spam taser shots, though, and tasers were available at roundstart. This is more comparable to a meathook, imo, except instead of hardstunning you, it has a chance to turn you into a syndicate assault cyborg.

**Comment by CRITAWAKETS** [comment]
> > (you can "just avoid them", but you can say that about taser shots and look where they are now)
> 
> You could spam taser shots, though, and tasers were available at roundstart. This is more comparable to a meathook, imo, except instead of stunning you, it has a chance to turn you into a syndicate assault cyborg.

except you can just fire again

oh and syndie assault borgs have free license to kill the crew and are stronk as fuk

**Comment by Shadowflame909** [comment]
> 
> 
> > > (you can "just avoid them", but you can say that about taser shots and look where they are now)
> > 
> > 
> > You could spam taser shots, though, and tasers were available at roundstart. This is more comparable to a meathook, imo, except instead of stunning you, it has a chance to turn you into a syndicate assault cyborg.
> 
> except you can just fire again
> 
> oh and syndie assault borgs have free license to kill the crew and are stronk as fuk

3 second cooldown

Willing polymorph doesn't create antags.

Imagine shooting an enemy with polymorph, and turning them into a meth speed xeno. That's what we call random backfiring.

This is just the one common methods of obtaining something insane. You can also get polymorph from the wabbajack shuttle. (If it's revs, then it'll last the entire round.)

You can get a pulse rifle from an arcade machine, or from a random floortile in space. A working legitimate d20 from a pride ruin in lavaland, or in maint.

All I'm saying is that such crazy "power" being in the crews hands is nothing new, and has always and will continue to exist.

End-Game medical can create romerol zombies freely after all.

**Comment by improvedname** [comment]
***Daily reminder that unwilling crewmembers hit by the bolt of change due policy of wizard rounds are turned into free-antags and pretty can murderbone anyone at that point.***

**Comment by Shadowflame909** [comment]
**Daily reminder that any havoc that comes from Xenobio leave the xenobiologist at fault.**

They're either an antag, or they're being permabanned.

That's how it worked for spiders. That's how it works for virologists who leave deadly viruses on counters for assistants to drink. That's how it works for scientists who throw bag of holdings a over the hallways.

Using the same tools mentioned before to say, the crew gets access to dangerous shiz. That doesn't mean we should remove them all. 

BoH removal or magicarp. The argument would still be the same.

**Comment by XDTM** [comment]
Even if the possibility of morphing the enemy into something strong could balance them in combat (and that's questionable, since even if a traitor is xeno'd or syndiborged, they still end up irreparably redtexting due to mobtype) there is the whole powergame aspect of making mobs to harvest materials from that is heavily problematic by itself and has little to no chance of backfiring if done properly.

**Comment by Shadowflame909** [comment]
Even on the unlikely chance sentient mobs would agree to be harvested. The time frame this would come into play is when the round has reached the tipping point or end of content.

If someone wants to harvest xenos past the hour mark. All power to them. Balance hasn't cared about that time frame. Or else they wouldn't have been obtained. Same with any other jobs top tier weaponry.

**Comment by XDTM** [comment]
If i'm not wrong rounds are getting pushed towards the 60-90 minutes average duration, which would make gold slimes and light pink slimes fairly easy to achieve before reaching end-of-content status.

Then again, it's good to have late-round powerful projects like these, but i'd rather take magicarps off and implement this in a proper way that doesn't rely on indiscriminately copying mechanics and mobs that were made for other purposes.

**Comment by wesoda25** [comment]
Just make the polymorph temporary, much better solution than complete removal

**Comment by OverlyGeneric** [comment]
Remove magicarp and replace it with more interesting if you're dead-set on this path.

**Comment by JusticeGoat** [comment]
Cripes the change carp is rare enough i think having some kinda chaos once in a while is fine.

**Comment by MCterra10** [comment]
Bruh

**Comment by Ty-the-Smonk** [comment]
Just remove the carp if you are doing this, if you can’t handle a little chaos then get rid of it

**Review by kriskog — APPROVED** [review]
Just remove the magicarps from the gold pool tbh, but this is yet another good step. I dont care about your antag round or whatever but magic should largely be kept to wizard rounds.

Any dedicated xenobiologist can push out a double digit amount of magicarps in no time at all.

**Comment by nullAdept** [comment]
indeed bruh

**Comment by Shadowflame909** [comment]
Says the gamer who added a legitimate wizard d20 into the maint spawn pools

**Comment by kriskog** [comment]
A 1/500 rounds spawn with 1/20 chance of converting the round into a wizard round is entirely different than something you can conceivably make every round.

**Comment by trollbreeder** [comment]
which is why i said make a slime crossbreed that polymorphs whoever uses it so we dont lose !!FUN!! value

**Comment by trollbreeder** [comment]
crossbreeds are tedious to make requiring 3 dead baby slimes at T4 processor and 4 slimes at T3 (which requires 2 generations of slime if you dont wanna lose that color) or a reproductive (which requires a metric fuggetonne of monkey cubes and even best case scenario you need 9 monkey cubes)

**Comment by kriskog** [comment]
We dont need to add a replacement before we remove something. You are however free to suggest such a thing in a PR of your own or on the ideas forum.

**Comment by cacogen** [comment]
what have we done

**Comment by WarlockD** [comment]
I just found this out.  Pointless and stupid.  A fun thing gone because somone got pissy over dieing on rng.
